js/adapt-contrib-xapi.js

```diff
@@ -64,1568 +61,1516 @@ define([
-          ADL.launch(_.bind(function(error, launchData, xapiWrapper) {
-            if (error) {
-              return callback(error);
-            }
+  /**
+   * Replace the hard-coded _learnerInfo data in _globals with the actual data from the LRS.
+   */
+  getLearnerInfo() {
+    const globals = Adapt.course.get('_globals');
 
-            // Initialise the xAPI wrapper.
-            this.xapiWrapper = xapiWrapper;
+    if (!globals._learnerInfo) {
+      globals._learnerInfo = {};
+    }
 
-            this.set({
-              actor: launchData.actor
-            });
+    _.extend(globals._learnerInfo, Adapt.offlineStorage.get('learnerinfo'));
+  }
 
-            this.xapiWrapper.strictCallbacks = true;
+  /**
+   * Intializes the ADL xapiWrapper code.
+   * @param {ErrorOnlyCallback} callback
+   */
+  initializeWrapper(callback) {
+    // If no endpoint has been configured, assume the ADL Launch method.
+    if (!this.getConfig('_endpoint')) {
+      //check to see if configuration has been passed in URL
+      this.xapiWrapper = window.xapiWrapper || ADL.XAPIWrapper;
+      if (this.checkWrapperConfig()) {
+        // URL had all necessary configuration so we continue using it.
+        // Set the LRS specific properties.
+        this.set({
+          registration: this.getLRSAttribute('registration'),
+          actor: this.getLRSAttribute('actor')
+        });
 
-            callback();
-          }, this), true, true);
-        }
-      } else {
-        // The endpoint has been defined in the config, so use the static values.
-        // Initialise the xAPI wrapper.
-        this.xapiWrapper = window.xapiWrapper || ADL.XAPIWrapper;
-
-        // Set any attributes on the xAPIWrapper.
-        var configError;
-        try {
-          this.setWrapperConfig();
-        } catch (error) {
-          configError = error;
-        }
+        this.xapiWrapper.strictCallbacks = true;
 
-        if (configError) {
+        callback();
+        return;
+      } 
+      // If no endpoint is configured, assume this is using the ADL launch method.
+      ADL.launch((error, launchData, xapiWrapper) => {
+        if (error) {
           return callback(error);
         }
 
-        // Set the LRS specific properties.
+        // Initialise the xAPI wrapper.
+        this.xapiWrapper = xapiWrapper;
+
         this.set({
-          registration: this.getLRSAttribute('registration'),
-          actor: this.getLRSAttribute('actor')
+          actor: launchData.actor
         });
 
         this.xapiWrapper.strictCallbacks = true;
 
         callback();
-      }
-    },
+      }, true, true);
+      return;
+    }
+    // The endpoint has been defined in the config, so use the static values.
+    // Initialise the xAPI wrapper.
+    this.xapiWrapper = window.xapiWrapper || ADL.XAPIWrapper;
+
+    // Set any attributes on the xAPIWrapper.
+    let configError;
+    try {
+      this.setWrapperConfig();
+    } catch (error) {
```

Comment:
This needs changed to
`catch (configError)`
and the variable declaration above should be removed